ap/selection/test.py

```diff
@@ -20,9 +20,28 @@ def extract(array, idx):
     return array
 
 
+# pseudo-selectors for declaring dependence on shifts
+
+@selector
+def jet_energy_shifts(events):
+    return events
+
+
+@jet_energy_shifts.update
+def update(self, config_inst, dataset_inst, **kwargs):
+    """Declare dependence on JEC/JER uncertainty shifts."""
+    self.shifts |= {
+        f"jec_{junc_name}_{junc_dir}"
+        for junc_name in config_inst.x.jec.uncertainty_sources
+        for junc_dir in ('up', 'down')
+    }
+    if dataset_inst.is_mc:
+        self.shifts |= {"jer_up", "jer_down"}
+
+
 # object definitions
 # TODO: Producer instead of Selector
-@selector(uses={"Jet.pt", "Jet.eta"})
+@selector(uses={"Jet.pt", "Jet.eta"}, shifts={jet_energy_shifts})
```

Comment:
Yeah, I see how this could become problematic with more calibrators. But I'm not sure the alternative to implement the shifts only for the main selector is better. In principle one should be able to use each selector independently, so each one should properly declare its shifts.

Maybe we could just leave it as it is for the moment and see if we run into problems once more selectors/calibrators are added?